sycl/include/CL/sycl/accessor.hpp

```diff
@@ -839,8 +839,17 @@ class __SYCL_SPECIAL_CLASS accessor :
     size_t Result = 0;
     // Unroll the following loop for both host and device code
     __SYCL_UNROLL(3)
-    for (int I = 0; I < Dims; ++I)
-      Result = Result * getMemoryRange()[I] + getOffset()[I] + Id[I];
+    for (int I = 0; I < Dims; ++I) {
+      Result = Result * getMemoryRange()[I] + Id[I];
+#if __cplusplus >= 201703L
+      if constexpr (!(PropertyListT::template has_property<
+                        sycl::ext::oneapi::property::no_offset>())) {
+        Result += getOffset()[I];
+      }
+#else
+      Result += getOffset()[I];
```

Comment:
Is there a reason to not have an if-statement here? Granted it cannot be `constexpr` but since the expression should evaluate to a constant the if-statement is optimized out anyway.

Reply:
@steffenlarsen has_property is only available with C++17. The whole extension is written agains SYCL 2020, and I never bothered supporting older versions. So, even if Nikita removes constexpr, we’d still need to guard C++17 usage, so that wouldn’t help us with anything.